consensus/enclave/impl/src/lib.rs

```diff
@@ -173,9 +181,17 @@ impl ConsensusEnclave for SgxConsensusEnclave {
         client_self_id: &ResponderId,
         sealed_key: &Option<SealedBlockSigningKey>,
         fee_map: &FeeMap,
+        block_version: BlockVersion,
     ) -> Result<(SealedBlockSigningKey, Vec<String>)> {
-        // Inject the fee into the peer ResponderId.
-        let peer_self_id = fee_map.responder_id(peer_self_id);
+        // Inject the fee map and block version into the peer ResponderId.
+        let peer_self_id = ResponderId::from_str(&format!(
+            "{}-block-version-{}",
+            fee_map.responder_id(peer_self_id),
+            block_version
+        ))
+        .unwrap();
```

Comment:
lmk if you guys think this responder-id part should be done differently

Reply:
Hmm, has this been tested on a deployed network?
If you update the responder id it should get updated in all other places below, which would result in a bunch of repetition of this string formatting stuff. For example, https://github.com/mobilecoinfoundation/mobilecoin/blob/8250b9ac242a72f025c2d77e6ffb53dba1832b8a/consensus/enclave/impl/src/lib.rs#L267

I think a better approach at this point would be to add a `responder_id` method to this object that takes a responder id and appends whatever it is we decided we need to append to it.